commit 0ca8c71db4a391546e7a720be75c1353e4b0ff0d
Author: Joshua Emele <jemele@acm.org>
Date:   Wed Nov 7 16:07:40 2018 -0800

    lan78xx: Debounce link events to minimize poll storm
    
    The bInterval is set to 4 (i.e. 8 microframes => 1ms) and the only bit
    that the driver pays attention to is "link was reset". If there's a
    flapping status bit in that endpoint data, (such as if PHY negotiation
    needs a few tries to get a stable link) then polling at a slower rate
    would act as a de-bounce.
    
    See: https://github.com/raspberrypi/linux/issues/2447

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 59bda07dc9d7..86d2f1d4c758 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -448,6 +448,11 @@ static bool enable_tso;
 module_param(enable_tso, bool, 0644);
 MODULE_PARM_DESC(enable_tso, "Enables TCP segmentation offload");
 
+#define INT_URB_MICROFRAMES_PER_MS	8
+static int int_urb_interval_ms = 8;
+module_param(int_urb_interval_ms, int, 0);
+MODULE_PARM_DESC(int_urb_interval_ms, "Override usb interrupt urb interval");
+
 static int lan78xx_read_reg(struct lan78xx_net *dev, u32 index, u32 *data)
 {
 	u32 *buf = kmalloc(sizeof(u32), GFP_KERNEL);
@@ -3816,7 +3821,12 @@ static int lan78xx_probe(struct usb_interface *intf,
 	dev->pipe_intr = usb_rcvintpipe(dev->udev,
 					dev->ep_intr->desc.bEndpointAddress &
 					USB_ENDPOINT_NUMBER_MASK);
-	period = dev->ep_intr->desc.bInterval;
+	if (int_urb_interval_ms <= 0)
+		period = dev->ep_intr->desc.bInterval;
+	else
+		period = int_urb_interval_ms * INT_URB_MICROFRAMES_PER_MS;
+
+	netif_notice(dev, probe, netdev, "int urb period %d\n", period);
 
 	maxp = usb_maxpacket(dev->udev, dev->pipe_intr, 0);
 	buf = kmalloc(maxp, GFP_KERNEL);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1450_0ca8c71db4a391546e7a720be75c1353e4b0ff0d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1450_0ca8c71db4a391546e7a720be75c1353e4b0ff0d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
